### PR TITLE
Obj_Bigicicle

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_Obj_Bigicicle/z_obj_bigicicle.c [hunk 1]
```diff
@@ -34,51 +35,287 @@ const ActorInit Obj_Bigicicle_InitVars = {
...
+};
+
+void ObjBigicicle_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjBigicicle* this = THIS;
+    f32 sp30;
+    s32 sp28;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+
+    sp28 = (this->actor.params >> 8) & 0xFF;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Bigicicle/z_obj_bigicicle.c:107]: macro

File: src/overlays/actors/ovl_Obj_Bigicicle/z_obj_bigicicle.c [hunk 2]
```diff
@@ -34,51 +35,287 @@ const ActorInit Obj_Bigicicle_InitVars = {
...
+    if ((sp28 == 0) || (sp28 == 0xFF)) {
+        sp28 = 0x3C;
+    }
+    Actor_SetScale(&this->actor, sp28 * 0.001f);
+
+    this->actor.params &= 0xFF;
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    sp30 = sp28 * 0.016666668f;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Bigicicle/z_obj_bigicicle.c:115]: `* (1.0f / 60.0f)` or maybe `/ 60.0f`?

File: src/overlays/actors/ovl_Obj_Bigicicle/z_obj_bigicicle.c [hunk 3]
```diff
@@ -34,51 +35,287 @@ const ActorInit Obj_Bigicicle_InitVars = {
...
+
+    Collider_DestroyCylinder(globalCtx, &this->collider1);
+    Collider_DestroyCylinder(globalCtx, &this->collider2);
+}
+
+void func_80AE8DE4(ObjBigicicle* this, GlobalContext* globalCtx) {
+    s32 i;
+    Vec3f sp98;
+    Vec3f sp8C;
+    f32 temp_f20 = this->actor.scale.x * 16.666666f;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Bigicicle/z_obj_bigicicle.c:158]: Maybe?
```suggestion
    f32 temp_f20 = this->actor.scale.x * (50.0f / 3.0f);
```

File: src/overlays/actors/ovl_Obj_Bigicicle/z_obj_bigicicle.c [hunk 4]
```diff
@@ -34,51 +35,287 @@ const ActorInit Obj_Bigicicle_InitVars = {
...
+
+        func_80AE8DE4(this, globalCtx);
+        this->unk_149++;
+        this->unk_14A = 50;
+        this->actionFunc = func_80AE9180;
+    }
+}
+
+void func_80AE9090(ObjBigicicle* this, GlobalContext* globalCtx) {
+    if (ActorCutscene_GetCanPlayNext(this->actor.cutscene) != 0) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Bigicicle/z_obj_bigicicle.c:197]:
```suggestion
    if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
```

File: src/overlays/actors/ovl_Obj_Bigicicle/z_obj_bigicicle.c [hunk 5]
```diff
@@ -34,51 +35,287 @@ const ActorInit Obj_Bigicicle_InitVars = {
...
+}
 
-extern ColliderCylinderInit D_80AE9820;
-extern ColliderCylinderInit D_80AE984C;
-extern InitChainEntry D_80AE9888[];
+void func_80AE939C(ObjBigicicle* this, GlobalContext* globalCtx) {
+    s32 i;
+    Vec3f sp98;
+    Vec3f sp8C;
+    f32 temp_f20 = this->actor.scale.x * 16.666666f;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Bigicicle/z_obj_bigicicle.c:270]: same here

**Review by hensldm — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**